src/api/registerForEvents.js

```diff
@@ -50,6 +50,18 @@ const transformCrossRealmBot = (
 
 const transform = (rawInitialData: RawInitialData, auth: Auth): InitialData => ({
   ...rawInitialData,
+
+  // Transform the newer `realm_linkifiers` format, if present, to the
+  // older `realm_filters` format. We do the same transformation on
+  // 'realm_linkifiers' events.
+  realm_filters: rawInitialData.realm_linkifiers
+    ? rawInitialData.realm_linkifiers.map(({ pattern, url_format, id }) => [
+        pattern,
+        url_format,
+        id,
+      ])
+    : rawInitialData.realm_filters,
```

Comment:
Do we want to make it so that if the server doesn't send either back, it's `[]`, or does that not matter?

Reply:
I think if the server didn't send either back, it would be a server bug (as long as we didn't suddenly stop requesting them for some reason). That might need some defensive code, but I'd be inclined to investigate that at the time it's discovered.